trin-core/src/portalnet/overlay.rs

```diff
@@ -376,6 +376,14 @@ where
                 .map(|node| node.value.enr())
                 .collect();
 
+            // Sort all eligible nodes by proximity to the content and take the closest k.
+            interested_enrs.sort_by(|a, b| {
+                let distance_a = XorMetric::distance(&content_key.content_id(), &a.node_id().raw());
+                let distance_b = XorMetric::distance(&content_key.content_id(), &b.node_id().raw());
+                distance_a.partial_cmp(&distance_b).unwrap()
```

Comment:
Is this a safe `unwrap`?

Reply:
@ogenev Good question, I was wondering the same. I think the answer is yes but I updated it to provide a default value and a warning in case it's not.